language/bytecode-verifier/src/verifier.rs

```diff
@@ -24,6 +25,7 @@ use move_binary_format::{
 /// minimize the code locations that need to be updated should a new checker
 /// is introduced.
 pub fn verify_module(module: &CompiledModule) -> VMResult<()> {
+    BoundsChecker::verify_module(&module).map_err(|e| e.finish(Location::Undefined))?;
```

Comment:
Can you comment here? It is actually unsafe to lookup the self_id for the location and would be good to document that